packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+    const eventData = evt.detail;
+    const element = eventData.element;
+    const config = this.configuration;
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine, viewport } = enabledElement;
+
+    this._commonData.viewportIdsToRender = [viewport.id];
+
+    const annotations =
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    this._commonData.mouseLocation = eventData.currentPoints.canvas;
+
+    if (this.isActive) {
+      annotations[this._commonData.selectedIndex].highlighted = true;
+    } else {
+      const canvasCoords = eventData.currentPoints.canvas;
+      const radius = this._distanceFromPoint(
+        viewport,
+        annotations[this._commonData.selectedIndex]?.data,
+        canvasCoords
+      );
+      if (radius > 0) {
+        this.toolSize = Math.min(config.maxToolSize, radius);
+      }
+    }
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      this._commonData.viewportIdsToRender
+    );
+  }
+
+  /**
+   * Returns interactable freehand ROI annotations
+   *
+   * @param element - The viewport element
+   */
+  _filterInteractableFreehandRoiAnnotationsForElement(element: HTMLDivElement) {
+    const config = this.configuration;
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngineId, viewportId } = enabledElement;
+    let interactableAnnotation = [];
+
+    const toolGroup = ToolGroupManager.getToolGroupForViewport(
+      viewportId,
+      renderingEngineId
+    );
+
+    const toolInstance = toolGroup.getToolInstance(config.referencedToolName);
+
+    config.referencedToolNames.forEach((referencedToolName: string) => {
+      const annotation = getAnnotations(referencedToolName, element);
+      if (annotation) {
+        interactableAnnotation = [...interactableAnnotation, ...annotation];
+      }
+    });
+
+    interactableAnnotation =
+      toolInstance.filterInteractableAnnotationsForElement(
+        element,
+        interactableAnnotation
+      );
+
+    return interactableAnnotation;
+  }
+
+  _configureToolSize(evt: EventTypes.InteractionEventType): void {
+    const config = this.configuration;
+
+    if (this.toolSize && config.maxToolSize) {
+      return;
+    }
+
+    const eventData = evt.detail;
+    const element = eventData.element;
+    const minDim = Math.min(element.clientWidth, element.clientHeight);
+    const sampleArea = Math.pow(minDim / 5, 2);
+    const maxRadius = Math.pow(sampleArea / Math.PI, 0.5);
+
+    this.toolSize = maxRadius;
+    config.maxToolSize = maxRadius;
+  }
+
+  /**
+   * Sculpts the freehand ROI with the circular freehandSculpter tool, moving,
+   * adding and removing handles as necessary.
+   *
+   * @param eventData - Data object associated with the event.
+   * @param points - Array of points
+   */
+  _sculpt(eventData: any, points: Array<Types.Point3>): void {
```

Comment:
Can you make this method protected sculpt( ...)
It may well be desirable to extend your class in the future.

Reply:
Updated